### PR TITLE
Add Github Actions

### PR DESCRIPTION
**Pull request by Luni-4**
This PR adds `GitHub Actions` for `cbindgen`

Thanks in advance for your review! :)

### CODE REVIEW

**Comment by emilio** [comment]
We have travis set up at https://travis-ci.org/eqrion/cbindgen... Can you elaborate on the reasoning for this?

**Comment by Luni-4** [comment]
Sure! Well, since `GitHub Actions` is well integrated in the `GitHub` environment, I thought it would be useful to add that kind of CI as well. In addition, the use of actions allow to simplify many tasks.

**Comment by Luni-4** [comment]
Are you still interested in this one, @emilio?

**Comment by emilio** [comment]
Hmm, not particularly excited about having two CI systems. We should have just one. I don't have a problem to switch to github actions if it's better, but I probably need admin on the repo or such which I don't have :)

**Comment by Luni-4** [comment]
>I don't have a problem to switch to github actions if it's better, but I probably need admin on the repo or such which I don't have :)

IIRC they should be enabled automatically, so they should run without setting up anything else

**Comment by Luni-4** [comment]
@emilio 

This PR is complete, it does the same stuff of the `travis` script and more:
- Run `cargo clippy` and `cargo fmt` on each commit for each push and pull request event
-  Deploy the `cbindgen` binary automatically just pushing the desired tag. At the same time, the `api_token` isn't shown up in the script like in `travis`
- Github Actions are perfectly integrated in the Github environment

**Review by emilio — COMMENTED** [review]
If we do this, `ci/script.sh` needs to be removed. Is there a way to confirm this works before merging somehow?

**Comment by Luni-4** [comment]
Yep, I will remove that! [Here](https://github.com/Luni-4/cbindgen/runs/1075468921) you can see that works

File: .github/workflows/cbindgen.yml
Comment (Luni-4) [reply .github/workflows/cbindgen.yml:37]: It's red because clippy warnings are transformed into errors. Perhaps we could fix those warnings alongside this PR or in an another one. Wdyt?

Comment (emilio) [reply .github/workflows/cbindgen.yml:37]: I'd prefer another one, but meanwhile this should probably be green.

Comment (Luni-4) [reply .github/workflows/cbindgen.yml:37]: Let's merge this PR after #571

**Comment by Luni-4** [comment]
Now it's [all](https://github.com/Luni-4/cbindgen/runs/1078499542) green! Yay!

**Comment by Luni-4** [comment]
Perfect, now the badge is fixed! :)

File: .github/workflows/deploy.yml
Comment (emilio) [inline .github/workflows/deploy.yml:30]: I'm not sure stripping is great. I'd rather have useful stacks tbh. But this is only for releases so we can change it in a follow-up.

**Review by emilio — APPROVED** [review]
Thank you for doing this!

**Pull request merged**
